pkg/utils/utils.go

```diff
@@ -52,6 +56,32 @@ func AreSelectorsSame(lhs, rhs map[string]string) bool {
 	return true
 }
 
+// ForceLinuxNodeSelector takes given selector map and returns a selector map with linux node selector added into it.
+// If there is already a node type selector and is different from "linux" then it is overridden to "linux" type.
+// See https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#interlude-built-in-node-labels
+func ForceLinuxNodeSelector(selectors map[string]string) map[string]string {
+	if os, ok := selectors[OS_NODE_LABEL]; ok {
+		if os == LINUX_NODE_LABEL_VALUE {
+			return selectors
+		}
+		// Can this not be "linux"?
+		logrus.Warnf("Unexpected node selector value: %s=%s", OS_NODE_LABEL, os)
+		return selectors
+	}
+	newSelectors := make(map[string]string)
+	if selectors == nil {
```

Comment:
this is redundant with the logic above. did you mean to `selectors = map[string]string{}` after instead?

Reply:
You are right the logic there was quite chaotic. Fixed.